src/vellum/workflows/outputs/base.py

```diff
@@ -204,13 +204,11 @@ def __eq__(self, other: object) -> bool:
         if not isinstance(other, dict):
             return super().__eq__(other)
 
-        outputs = {
-            name: value for name, value in vars(self).items() if not name.startswith("_") and value is not undefined
-        }
+        outputs = {ref.name: value for ref, value in self if value is not undefined}
```

Comment:
I want to remove the need to do ` if value is not undefined` one day, but still need to audit which use cases that negatively affects